src/include/mallocMC/mallocMC_utils.hpp

```diff
@@ -111,14 +59,26 @@ namespace mallocMC
         using type = unsigned long long;
     };
 
+#if defined(__HIP__) || defined(_CUDA_ARCH_)
```

Comment:
it must be `__CUDA_ARCH__` everywhere.

Reply:
and `__HIP_DEVICE_COMPILE__== 1 && defined(__HIP__)` to check if you are on a HIP device.

https://github.com/ROCm-Developer-Tools/HIP/blob/master/docs/markdown/hip_porting_guide.md#compiler-defines-summary

`__HIP_DEVICE_COMPILE__` is undefined on the host side

Reply:
your link suggests that just checking if `__HIP_DEVICE_COMPILE__` is defined is enough.